runtime/router/router.go

```diff
@@ -153,3 +159,20 @@ func (r *Router) allowed(path, reqMethod string) string {
 
 	return strings.Join(allow, ", ")
 }
+
+func (r *Router) isWhitelistedPath(path string) bool {
```

Comment:
lets use fixed path withtout path param in whitelist. santize those urls on init. 

This way it will be much simpler.

Reply:
It will not work for get because the variable path param gets changed for each request

Reply:
Saving isWhitelisted at the leaf node does not work for the paths such as "/a/b/c" and "/a/:var/c". We discussed this in our zoom call and agreed to not store this flag in the trie. I do not see any issue with using this method as it looks cleaner and easier to understand IMHO.

Reply:
If we store at leaf node I am not convinced now why it would not work. 
1. Without whitelist, it would have conflicted.
2. As the concrete path is whitelisted we should allow its storing.
3. For any concrete path not whitelisted we should detect the conflict as in previous case.

Reply:
Also, the option of storing and encoding all the metadata including whitelist info is generally a better option. Coupling it with getter to again consult the metatdata is not considered clean. However I will leave it you to decide what you think is better.

Reply:
Saving it in whitelist does not work for all the use cases and it cause other issues. I will go with approach as this looks cleaner and easier to maintain IMO.